sandbox/gpio/common.go

```diff
@@ -125,9 +126,23 @@ var lockAggregator = func() (unlocker func(), err error) {
 	}, nil
 }
 
+var kmodLoadModule = kmod.LoadModule
 var aggregatorCreationTimeout = 120 * time.Second
 
 func addAggregatedChip(ctx context.Context, sourceChip *ChardevChip, lines strutil.Range) (chip *ChardevChip, err error) {
+	// Make sure the gpio-aggregator module is loaded because the
+	// systemd security backend comes before the kmod security
+	// backend, there is an edge case on first connection where
+	// the helper service could be started before the gpio-aggregator
+	// module is loaded.
+	if _, err := osStat(filepath.Join(dirs.GlobalRootDir, aggregatorDriverDir)); errors.Is(err, os.ErrNotExist) {
+		if err := kmodLoadModule("gpio-aggregator", nil); err != nil {
+			return nil, err
+		}
+	} else if err != nil {
+		return nil, err
+	}
+
```

Comment:
In principle this is correct, but I'm wondering if maybe we could do the flow more explicitly in snap-gpio-helper in the export path. 

What if we move this part into something like gpio.EnsureAggregatorDriver() and call it explicitly from export command handler right before gpio.ExportGadgetChardevChip ?

Reply:
I like this better, updated. Thanks!